code/client/src/components/LocalExport.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import protobuf from 'protobufjs/light'
+import message from '../message'
+import storage from '../storage'
+import { Button } from 'antd'
+import { ExportOutlined, LoadingOutlined } from '@ant-design/icons'
+import { useState } from 'react'
+
+const Field = protobuf.Field
+
+export function LocalExportMessage (properties) {
+  protobuf.Message.call(this, properties)
+}
+
+Field.d(1, 'string')(LocalExportMessage.prototype, 'wallet')
+Field.d(2, 'bytes', 'repeated')(LocalExportMessage.prototype, 'layers')
+
+const LocalExport = ({ wallet }) => {
+  const [loading, setLoading] = useState(false)
+
+  const handleExport = async () => {
+    try {
+      setLoading(true)
+
+      const layers = await storage.getItem(wallet.root)
+      const filename = `1wallet-${wallet.name.toLowerCase().split(' ').join('-')}.txt`
+      const msg = new LocalExportMessage({ wallet: JSON.stringify(wallet), layers: layers })
+      const buffer = LocalExportMessage.encode(msg).finish()
+      const blob = new Blob([buffer], { type: 'text/plain' })
```

Comment:
I think it should be `application/octet-stream` since it is binary data

Reply:
It seems like the file extension is affecting the mime-type, especially for Safari. I tried to use .pb or other extension, it ended up the mime type is empty.